src/python/pants/python/python_setup.py

```diff
@@ -167,6 +167,14 @@ def requirement_constraints(self) -> Optional[str]:
     def interpreter_search_paths(self):
         return self.expand_interpreter_search_paths(self.get_options().interpreter_search_paths)
 
+    def interpreter_search_paths_ensure_directories(self) -> List[str]:
```

Comment:
Preferred to return `Tuple[str, ...]`, I think. Even though it's not idiomatic in normal Python, tuple sequences are more useful than list sequences in V2 code.

Reply:
I was thinking that this is by definition full of absolute paths and is unlikely to be consumed directly by V2 code. The `List[str]` distinction seems to help make that more clear. It can be a tuple if we want.

Reply:
Got it. That's reasonable. Thanks!